adapters/repos/db/migrator.go

```diff
@@ -685,8 +685,16 @@ func (m *Migrator) RecalculateVectorDimensions(ctx context.Context) error {
 
 	// Iterate over all indexes
 	for _, index := range m.db.indices {
+		err := index.ForEachShard(func(name string, shard ShardLike) error {
+			return shard.resetDimensionsLSM()
+		})
+		if err != nil {
+			m.logger.WithField("action", "reindex").WithField("error", err).Warn("could not reset vector dimensions")
```

Comment:
I think that there's a dedicated method `WithError` for errors which you could use instead of `WithField("error", err)`